jumanji/environments/routing/pacman/env.py

```diff
@@ -0,0 +1,551 @@
+    state, timestep = jax.jit(env.reset)(key)
+    env.render(state)
+    action = env.action_spec().generate_value()
+    state, timestep = jax.jit(env.step)(state, action)
+    env.render(state)
+    ```
+    """
+
+    def __init__(
+        self,
+        generator: Optional[Generator] = None,
+        viewer: Optional[Viewer[State]] = None,
+        time_limit: Optional[int] = None,
+    ) -> None:
+        """Instantiates a `Pac-Man` environment.
+
+        Args:
+            generator: `Generator` whose `__call__` instantiates an environment instance.
+                Implemented options are [`AsciiGenerator`].
+            time_limit: the time_limit of an episode, i.e. the maximum number of environment steps
+                before the episode terminates. By default, set to 1000.
+            viewer: `Viewer` used for rendering. Defaults to `PacManViewer`.
+        """
+
+        self.generator = generator or AsciiGenerator(DEFAULT_MAZE)
+        self.x_size = self.generator.x_size
+        self.y_size = self.generator.y_size
+        self.pellet_spaces = self.generator.pellet_spaces
+        self._viewer = viewer or PacManViewer("Pacman", render_mode="human")
+        self.time_limit = 1000 or time_limit
+
+    def observation_spec(self) -> specs.Spec[Observation]:
+        """Specifications of the observation of the `PacMan` environment.
+
+        Returns:
+            Spec containing all the specifications for all the `Observation` fields:
+            - player_locations: tree of BoundedArray (int32) of shape ().
+            - grid: BoundedArray (int)) of the ingame maze with walls.
+            - ghost_locations: jax array (int) of ghost positions.
+            - power_up_locations: jax array (int) of power-pellet locations
+            - pellet_locations: jax array (int) of pellet locations.
+            - action_mask: jax array (bool) defining current actions.
+            - frightened_state_time: int counting time remaining in
+                scatter mode.
+            - score: (int) of total score obtained by player.
+        """
+        player_locations = specs.Spec(
+            Position,
+            "PositionSpec",
+            y=specs.BoundedArray((), jnp.int32, 0, self.x_size - 1, "y_coordinate"),
+            x=specs.BoundedArray((), jnp.int32, 0, self.y_size - 1, "x_coordinate"),
+        )
+        grid = specs.BoundedArray(
+            shape=(self.x_size, self.y_size),
+            dtype=jnp.int32,
+            name="grid",
+            minimum=0,
+            maximum=1,
+        )
+        ghost_locations = specs.Array(
+            shape=(4, 2),
+            dtype=jnp.int32,
+            name="ghost_locations",
+        )
+        power_up_locations = specs.Array(
+            shape=(4, 2),
+            dtype=jnp.int32,
+            name="power_up_locations",
+        )
+        pellet_locations = specs.Array(
+            shape=(self.pellet_spaces.shape[0], self.pellet_spaces.shape[1]),
```

Comment:
I am confused, it seems that `pellet_spaces = jnp.array([0, 0])`, which would mean that `pellet_spaces.shape[1]` does not exist. If we are talking about locations, shouldn't `pellet_locations` be of shape (..., 2)?